anaconda_project/internal/notebook_analyzer.py

```diff
@@ -0,0 +1,55 @@
+# -*- coding: utf-8 -*-
+# ----------------------------------------------------------------------------
+# Copyright © 2017, Continuum Analytics, Inc. All rights reserved.
+#
+# The full license is in the file LICENSE.txt, distributed with this software.
+# ----------------------------------------------------------------------------
+"""Analyze notebook files."""
+from __future__ import absolute_import
+
+import codecs
+import json
+import re
+
+from anaconda_project.internal.py2_compat import is_string
+
+_comment_re = re.compile("#.*$", re.MULTILINE)
+_fusion_register_re = re.compile(r"^\s*@fusion\.register", re.MULTILINE)
+
+
+# see if some source has @fusion.register. This is
+# obviously sort of heuristic, but without executing
+# the python we can only do so much.
+def _has_fusion_register(source):
+    # dump comments so commenting out fusion.register
+    # would work as expected
+    source = re.sub(_comment_re, "", source)
+    return re.match(_fusion_register_re, source) is not None
```

Comment:
These regexes are how we decide whether to automatically set `registers_fusion_function`. Of course, the developer can change what we set by default if we get it wrong.